server/index.js

```diff
@@ -54,6 +54,100 @@ app.get('/api/products/:productId', (req, res, next) => {
     .catch(err => next(err));
 });
 
+app.get('/api/cart', (req, res, next) => {
+  const cartId = req.session.cartId;
+  if (!req.session.cartId) {
+    return res.json([]);
+  } else {
+    const sql = `
+  select "c"."cartItemId",
+       "c"."price",
+       "p"."productId",
+       "p"."image",
+       "p"."name",
+       "p"."shortDescription"
+  from "cartItems" as "c"
+  join "products" as "p" using ("productId")
+ where "c"."cartId" = $1
+  `;
+    const params = [cartId];
+    db.query(sql, params)
+      .then(result => res.json(result.rows));
+  }
+});
+
+app.post('/api/cart', (req, res, next) => {
+  const productId = Number(req.body.productId);
+  if (!Number.isInteger(productId) || productId <= 0) {
+    return res.status(400).json({ error: '"productId" must be a postive integer' });
+  }
+  const sql = `
+  select "price"
+  from "products"
+  where "productId" = $1
+`;
+  const params = [productId];
+  db.query(sql, params)
+    .then(result => {
+      const product = result.rows[0];
+      const price = product.price;
```

Comment:
if there is no product with the chosen id, `product` will be undefined and you'll crash at line 93.